schematic/manifest/generator.py

```diff
@@ -82,8 +82,19 @@ def __init__(
 
         # Determine whether current data type is file-based
         is_file_based = False
-        if self.root:
+
+        if self.root:    
```

Comment:
Okay so I've reworked the logic in that area, and also updated the tests to now check for this new `ValueError` as well. Could you take a look @mialy-defelice ?

Reply:
So it seems like another test is actually checking to make sure that `None` is allowed to be passed in for `root`. It might be better to separate out this other change from the new error message @mialy-defelice